proto/api.proto

```diff
@@ -412,6 +423,39 @@ service Api {
     };
   }
 
+  rpc GetOrCreateUser(GetOrCreateUserRequest) returns (GetOrCreateUserResponse) {
```

Comment:
this endpoint seems a bit overloaded. consider splitting it into two? `Get` => either user or empty, and `PostCreateUserRequest`?